wordpress-mobile/WordPress-iOS
# Add search capability to the Site Intent Verticals screen

Fixes #18108

This PR adds the ability to search through verticals in the Site Intent Verticals screen. If the search criteria don't match any available vertical, users are still able to submit a custom name, which will appear at the top of the list, and proceed with the design selection. See animation below and issue #18108 for more details.

This PR also updates the feature flag for the Site Intent Question feature.

<p align=center>
<img height=450 src=https://user-images.githubusercontent.com/34376330/161312740-bfa178e0-457d-4b69-ac56-bafb83f031af.gif>
</p>



**KNOWN ISSUES**
- The default verticals appear on top when the full list is shown, this is not an issue per se, but we might need to change it.

To test:

Note: since we are using A/B testing, to make testing easier, you might want to comment the following code, around line 18 in `SiteIntentStep`:

`guard FeatureFlag.siteIntentQuestion.enabled && variant == .treatment else {
            return nil
        }`
This would also disable the feature flag check.


- checkout this branch and build/run the app
- log in with a valid wp.com account (if not already logged in)
- if the account has no sites, tap on "Create a new site", if the account has sites, tap on the site selection button then tap "+"
- select "Create WordPress.com site"
- make sure the search bar is visible and displays the placeholder "Eg. Fashion, Poetry, Politics"
- also make sure that the default verticals are displayed
- tap in the search field to initiate search
- make sure that you see the tracking event in the console log:
`🔵 Tracked: enhanced_site_creation_intent_question_search_focused <>`
- make sure that the list of verticals contains other entries than the defaults
- start typing
- make sure the list is filtered, consistently with the search criteria, and that at the top of the list, you see the option to add a custom vertical (with a plus icon, see animation above)
- Tap on the custom vertical (with the plus) and make sure that it navigates to the design selection screen
- Also make sure to see the following tracked event in the console:
`🔵 Tracked: enhanced_site_creation_intent_question_custom_vertical_selected <vertical_slug: foo>` where `foo` is the custom text you inputed
- Make sure that, if you input a name that matches exactly one of the available verticals, the custom vertical row does not appear.

## Regression Notes
1. Potential unintended areas of impact
None other than the site intent selection screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
test the feature on a variety of devices

3. What automated tests I added (or what prevented me from doing so)
none

PR submission checklist:

- [x] ~~I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.~~ Since we will A/B test, we won't update release notes just yet.
